collector/watchdog.go

```diff
@@ -0,0 +1,133 @@
+// Copyright 2023 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:build linux && !nowatchdog
+// +build linux,!nowatchdog
+
+package collector
+
+import (
+	"fmt"
+
+	"github.com/go-kit/log"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/procfs/sysfs"
+)
+
+type watchdogCollector struct {
+	fs     sysfs.FS
+	logger log.Logger
+}
+
+func init() {
+	registerCollector("watchdog", defaultDisabled, NewWatchdogCollector)
```

Comment:
This seems low cardinality enough, and useful enough, we could consider enabling this by default.

Reply:
Thanks, I have submitted a PR for this. #2953